### PR TITLE
Bjorncs/annotations [run-systemtest]

### PR DESCRIPTION
**Pull request by bjorncs**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: annotations/pom.xml [hunk 1]
```diff
@@ -46,6 +46,31 @@
...
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>build-helper-maven-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>parse-version</id>
+            <goals>
+              <goal>parse-version</goal>
```
Comment (jonmv) [inline annotations/pom.xml:56]: There's a plugin for everything, huh?

**Review by jonmv — APPROVED** [review]
What provides the bundle, though?

**Comment by jonmv** [comment]
There's no `annotations` bundle started in the container in that system test.

**Comment by bjorncs** [comment]
@jonmv It's embedded and re-exported through container-disc bundle. It's a common pattern in the container.

**Comment by bjorncs** [comment]
Fails on system test (which is still broken).

File: fileacquirer/pom.xml [hunk 2]
```diff
@@ -22,6 +22,12 @@
       <artifactId>config</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
+      <groupId>com.yahoo.vespa</groupId>
+      <artifactId>annotations</artifactId>
+      <version>${project.version}</version>
+      <scope>provided</scope>
```
Comment (gjoranv) [inline fileacquirer/pom.xml:29]: fileacquirer is a regular jar, so provided does not make much sense.

File: jdisc_core/pom.xml [hunk 3]
```diff
@@ -167,6 +167,12 @@
             <version>${project.version}</version>
             <scope>compile</scope>
         </dependency>
+        <dependency>
+            <groupId>com.yahoo.vespa</groupId>
+            <artifactId>annotations</artifactId>
+            <version>${project.version}</version>
+            <scope>provided</scope>
```
Comment (gjoranv) [inline jdisc_core/pom.xml:174]: 'provided' deps do not make sense for jdisc_core, as it's on the classpath. For this to be usable, it must be compile, and the packages re-exported via jdisc_core.jar.

**Comment by gjoranv** [comment]
Why is it necessary to export these annotations? Not necessarily wrong, but if we do it, they should be "exported" via jdisc_core, not container-disc.

**Comment by bjorncs** [comment]
> Why is it necessary to export these annotations? Not necessarily wrong, but if we do it, they should be "exported" via jdisc_core, not container-disc.

It's exported for the `@Inject` annotation.

Why is it required exporting it from jdisc_core instead of container-disc? It's the DI framework of container-disc that understands the new annotation, not Guice/jdisc_core.

**Comment by gjoranv** [comment]
It would be needed for user bundles that are not built with our bundle-plugin, which should be possible, of course. Apart from that, Inject does not really need to be exported.

These annotations were always embedded in jdisc_core and available via the classpath, so exporting from container-disc is a false flag.

**Comment by gjoranv** [comment]
I can fix the latter when finished with #22483

**Pull request merged**